core/src/main/java/com/exadel/aem/toolkit/core/filters/PagePolicyWriterFilter.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exadel.aem.toolkit.core.filters;
+
+import com.day.cq.wcm.api.Page;
+import com.day.cq.wcm.api.components.Component;
+import com.day.cq.wcm.api.components.ComponentEditConfig;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.api.request.RequestPathInfo;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.apache.sling.engine.EngineConstants;
+import org.apache.sling.jcr.resource.api.JcrResourceConstants;
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.injectorspecific.ValueMapValue;
+import org.osgi.framework.Constants;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@org.osgi.service.component.annotations.Component(
+        property = {
+                Constants.SERVICE_RANKING + ":Integer=-2147483648",
+                EngineConstants.SLING_FILTER_SCOPE + "=" + EngineConstants.FILTER_SCOPE_COMPONENT,
+                "sling.filter.resourceTypes" + "=" + "cq/gui/components/siteadmin/admin/page/winmode",
+                EngineConstants.SLING_FILTER_PATTERN + "=" + "/editor.*"
+        }
+)
+public class PagePolicyWriterFilter implements Filter {
+
+    private static final String META_TAG_FORMAT = "<meta name=\"cq:template\" content=\"%s\">";
+    private static final int LENGTH_TO_TRIM = "Granite.PolicyResolver.build('".length();
+    private static final String SCRIPT_FORMAT = "<script>%s</script>";
+
+    @Override
+    public void init(FilterConfig filterConfig) throws ServletException {
+
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse, FilterChain filterChain) throws IOException, ServletException {
```

Comment:
Consider splitting this method , it looks like extracting long .map() chains can improve readability, imho